### PR TITLE
fix: Input simulation no longer hangs when Editor updates stall

### PR DESCRIPTION
**Pull request by hatayama**
## Summary
- Input simulation commands now return a timeout instead of hanging when Unity Editor or Input System updates stop advancing.
- Frame-based waits are separated from wall-clock timeouts so cleanup and recovery remain predictable.

## User Impact
- Keyboard and mouse simulation commands no longer leave the CLI waiting forever when Unity stops producing the update they need.
- Timeout cleanup is queued safely on the Unity main thread, reducing stuck input state after interrupted simulations.
- Debug-break and input-simulation guidance now points users toward paused-frame verification for gameplay checks.

## Changes
- Removed the old EditorDelay primitive and replaced frame-dependent waits with EditorFrameWaiter.
- Reworked TimerDelay as a wall-clock timer and used it for timeout guards.
- Hardened keyboard and mouse simulation timeout, cancellation, and cleanup paths.
- Updated focused tests for frame waits, timeout cleanup, stale callback removal, and simulate input cancellation.

## Verification
- cli/dist/darwin-arm64/uloop compile --project-path "$(git rev-parse --show-toplevel)"
- cli/dist/darwin-arm64/uloop run-tests --project-path "$(git rev-parse --show-toplevel)" --test-mode PlayMode --filter-type regex --filter-value "(ApplyOnNextConfiguredUpdate_CancellationBeforeInputUpdate_ShouldRemoveCallback|Press_Cancellation_Should_ClearPressOverlay|KeyUp_CancellationAfterRelease_Should_ClearHeldState|WaitForRuntimeFrames_WhenFrameGoalCannotComplete_ShouldReturnTimedOut)"
- cli/dist/darwin-arm64/uloop run-tests --project-path "$(git rev-parse --show-toplevel)" --test-mode PlayMode --filter-type regex --filter-value "io.github.hatayama.UnityCliLoop.Tests.PlayMode.SimulateKeyboardTests"
- cli/dist/darwin-arm64/uloop run-tests --project-path "$(git rev-parse --show-toplevel)" --test-mode PlayMode --filter-type regex --filter-value "io.github.hatayama.UnityCliLoop.Tests.PlayMode.SimulateMouseInputTests"
- cli/dist/darwin-arm64/uloop run-tests --project-path "$(git rev-parse --show-toplevel)" --test-mode EditMode --filter-type exact --filter-value "io.github.hatayama.UnityCliLoop.Tests.Editor.EditorFrameWaiterTests"

### CODE REVIEW

File: Packages/src/Editor/ToolContracts/TimerDelay.cs [hunk 1]
```diff
@@ -1,119 +1,137 @@
...
+                    "WaitThenExecuteOnMainThread must start from Unity's main-thread synchronization context.");
             }
 
-            TaskCompletionSource<bool> tcs = new();
-            
-            Timer timer = null;
-            timer = new Timer(_ =>
+            await Wait(milliseconds, ct);
+            action();
+        }
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Editor/ToolContracts/TimerDelay.cs:67]: Thanks. I checked this path against C# await behavior: `WaitThenExecuteOnMainThread` awaits `Wait(milliseconds, ct)` without `ConfigureAwait(false)`, so the continuation captures and resumes on the caller's `SynchronizationContext`. `RunContinuationsAsynchronously` prevents inline continuation execution from `SetResult`, but it does not discard the captured context.

I added regression coverage in f17587a0 to assert that the delayed action runs on the captured `SynchronizationContext` and thread. No production change is needed here.

File: Packages/src/Editor/FirstPartyTools/Common/InputSystem/InputSystemUpdateHelper.cs [hunk 2]
```diff
@@ -31,47 +34,119 @@ internal static class InputSystemUpdateHelper
...
+                    if (Interlocked.CompareExchange(
+                            ref applyWaitState,
+                            ApplyWaitStateFinishedWithoutApply,
+                            ApplyWaitStateWaiting) == ApplyWaitStateWaiting)
+                    {
+                        subscription.Dispose();
+                        return InputSimulationWaitOutcome.TimedOut;
+                    }
+
+                    InputSimulationWaitOutcome appliedOutcome = await tcs.Task.ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Editor/FirstPartyTools/Common/InputSystem/InputSystemUpdateHelper.cs:143]: Valid. Fixed in e4e40975 by converting `apply()` failures into a faulted returned task, so the timeout path cannot wait forever on an orphaned `TaskCompletionSource`. I also added a PlayMode regression test covering the fault-and-cleanup behavior.

File: Packages/src/Editor/FirstPartyTools/SimulateMouseInput/SimulateMouseInputUseCase.cs [hunk 3]
```diff
@@ -177,40 +177,69 @@ private async Task<UnityCliLoopMouseInputSimulationResult> ExecuteClick(
...
+                        await ReleaseButtonIfPossible(mouse, button).ConfigureAwait(false);
+                    if (releaseOutcome == InputSimulationWaitOutcome.TimedOut)
+                    {
+                        waitOutcome = InputSimulationWaitOutcome.TimedOut;
+                        ScheduleTimedOutButtonCleanup(mouse, button, false);
+                    }
+                    else
+                    {
+                        await InputSystemUpdateHelper.SwitchToMainThreadIfNeeded(CancellationToken.None);
+                        MouseInputState.SetButtonUp(button);
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Editor/FirstPartyTools/SimulateMouseInput/SimulateMouseInputUseCase.cs:224]: I checked this one and did not change production code. The cleanup paths already await `InputSystemUpdateHelper.SwitchToMainThreadIfNeeded(CancellationToken.None)` immediately before `MouseInputState.SetButtonUp(button)` in both the release-success and not-applied branches. The non-timeout overlay cleanup runs after those switches, so this path is already back on the Unity main thread.

File: Packages/src/Editor/FirstPartyTools/SimulateMouseInput/Skill/SKILL.md [hunk 4]
```diff
@@ -48,9 +48,9 @@ uloop simulate-mouse-input --action <action> [options]
 | `SmoothDelta` | Mouse.current.delta (per-frame) | Inject mouse delta smoothly over `--duration` seconds (human-like camera pan) |
 | `Scroll` | Mouse.current.scroll | Inject scroll wheel input |
 
-### Optional Debug Break Inspection
+### Debug Break Inspection (Standard for E2E)
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Editor/FirstPartyTools/SimulateMouseInput/Skill/SKILL.md:51]: Valid. Fixed in e4e40975 by updating the workflow step so it matches the section heading and treats Debug Break inspection as the standard frame proof when mouse input verifies a state transition.

File: Packages/src/Editor/FirstPartyTools/SimulateMouseUi/Skill/SKILL.md [hunk 5]
```diff
@@ -76,9 +76,9 @@ uloop simulate-mouse-ui --action <action> --x <x> --y <y> [options]
 - `--bypass-raycast` still uses coordinates for pointer event positions, but chooses the clicked, long-pressed, or dragged GameObject by `--target-path`
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Editor/FirstPartyTools/SimulateMouseUi/Skill/SKILL.md:76]: Valid. Fixed in e4e40975 by updating the workflow step so it matches the section heading and treats Debug Break inspection as the standard frame proof when UI input verifies a state transition.

File: Packages/src/Editor/ToolContracts/EditorFrameWaiter.cs [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+        {
+            Debug.Assert(frameCount >= 0, "frameCount must be non-negative");
+            ct.ThrowIfCancellationRequested();
+
+            if (frameCount <= 0)
+            {
+                return Task.CompletedTask;
+            }
+
+            TaskCompletionSource<bool> completionSource = new TaskCompletionSource<bool>();
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Editor/ToolContracts/EditorFrameWaiter.cs]: Valid. Fixed in e4e40975 by creating the frame-wait `TaskCompletionSource` with `RunContinuationsAsynchronously`. I also added an EditMode regression test that uses a synchronous continuation to verify it no longer runs inline from Editor update.

File: Packages/src/Editor/ToolContracts/TimerDelay.cs [hunk 7]
```diff
@@ -1,119 +1,137 @@
...
+                throw new InvalidOperationException(
+                    "WaitThenExecuteOnMainThread must start from Unity's main-thread synchronization context.");
             }
 
-            TaskCompletionSource<bool> tcs = new();
-            
-            Timer timer = null;
-            timer = new Timer(_ =>
+            await Wait(milliseconds, ct);
+            action();
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Editor/ToolContracts/TimerDelay.cs:66]: Valid. Fixed in e4e40975 by waiting for one Editor update after the wall-clock delay before invoking the main-thread action. I also added an EditMode regression test proving a zero-millisecond delay no longer invokes the action inline.
